### ISSUE TITLE
count:poisson trouble

### ISSUE THREAD

**Issue by hectep** [dmlc/xgboost#396]
I've got this script running on kaggle scripts okay, but trying to run it on my local PC gives me the next error:
This application has requested the Runtime to terminate it in an unusual way.
Please contact the application's support team for more information.
Process finished with exit code -529697949

If i change the objective to reg:linear, its working as it should be. Is there some problem with xgboost or my script, or its not even related to xgboost? (python 2.7.10 , pycharm 4.5.3)

import pandas as pd
import numpy as np 
from sklearn import preprocessing
import xgboost as xgb
from sklearn.feature_extraction import DictVectorizer

def xgboost_pred(train,labels,test):
	params = {}
	params["objective"] = "count:poisson"
	params["eta"] = 0.01
	params["min_child_weight"] = 5
	params["subsample"] = 0.8
	params["colsample_bytree"] = 0.8
	params["scale_pos_weight"] = 1.0
	params["silent"] = 1
	params["max_depth"] = 8

	plst = list(params.items())	
	offset = 4000
	num_rounds = 10000
	xgtest = xgb.DMatrix(test)

	#create a train and validation dmatrices 
	xgtrain = xgb.DMatrix(train[offset:,:], label=labels[offset:])
	xgval = xgb.DMatrix(train[:offset,:], label=labels[:offset])

	#train using early stopping and predict
	watchlist = [(xgtrain, 'train'),(xgval, 'val')]
	model = xgb.train(plst, xgtrain, num_rounds, watchlist, early_stopping_rounds=50)
	preds1 = model.predict(xgtest)

	train = train[::-1,:]
	labels = np.log(labels[::-1])

	xgtrain = xgb.DMatrix(train[offset:,:], label=labels[offset:])
	xgval = xgb.DMatrix(train[:offset,:], label=labels[:offset])

	watchlist = [(xgtrain, 'train'),(xgval, 'val')]
	model = xgb.train(plst, xgtrain, num_rounds, watchlist, early_stopping_rounds=50)
	preds2 = model.predict(xgtest)
	
	preds = preds1*2.6 + preds2*7.4
	return preds

#load train and test 
train  = pd.read_csv('C:/Rdata/liberty/train.csv', index_col=0)
test  = pd.read_csv('C:/Rdata/liberty/test.csv', index_col=0)

labels = train.Hazard
train.drop('Hazard', axis=1, inplace=True)

train_s = train
test_s = test

train_s.drop('T2_V10', axis=1, inplace=True)
train_s.drop('T2_V7', axis=1, inplace=True)
train_s.drop('T1_V13', axis=1, inplace=True)
train_s.drop('T1_V10', axis=1, inplace=True)

test_s.drop('T2_V10', axis=1, inplace=True)
test_s.drop('T2_V7', axis=1, inplace=True)
test_s.drop('T1_V13', axis=1, inplace=True)
test_s.drop('T1_V10', axis=1, inplace=True)

columns = train.columns
test_ind = test.index
train_s = np.array(train_s)
test_s = np.array(test_s)

for i in range(train_s.shape[1]):
    lbl = preprocessing.LabelEncoder()
    lbl.fit(list(train_s[:,i]) + list(test_s[:,i]))
    train_s[:,i] = lbl.transform(train_s[:,i])
    test_s[:,i] = lbl.transform(test_s[:,i])

train_s = train_s.astype(float)
test_s = test_s.astype(float)
preds1 = xgboost_pred(train_s,labels,test_s)

train = train.T.to_dict().values()
test = test.T.to_dict().values()

vec = DictVectorizer()
train = vec.fit_transform(train)
test = vec.transform(test)

preds2 = xgboost_pred(train,labels,test)
preds = 0.6 * preds1 + 0.4 * preds2

#generate solution
preds = pd.DataFrame({"Id": test_ind, "Hazard": preds})
preds = preds.set_index('Id')
preds.to_csv('xgb lin300 + stdfull.csv')

**Comment by walterreade** [comment]
Out of curiosity, why are you fitting a poisson model to non-integer labels? (You are taking the log of the labels.)

**Comment by tqchen** [comment]
This could due to your MSVC version, see https://github.com/dmlc/xgboost/blob/master/src/utils/math.h#L34

Please do check if you have latest version of xgboost installed. It was supposed to throw an exception as opposed to terminate
